ping/ping.c

```diff
@@ -84,6 +85,12 @@ ping_func_set_st ping4_func_set = {
 #define	NROUTES		9		/* number of record route slots */
 #define TOS_MAX		255		/* 8-bit TOS field */
 
+/* max. IPv4 packet size - IPv4 header size - ICMP header size */
+#define ICMP_MAX_DATALEN (65535 - 20 - 8)
+
+/* max. IPv6 payload size - ICMPv6 Echo Reply Header */
+#define ICMPV6_MAX_DATALEN (65535 - sizeof (struct icmp6_hdr))
+
```

Comment:
Shouldn't we also lower the MAXPACKET now to 65535? We declare it on the stack almost twice as big, fill it with data, buf never use the upper half after this change.